assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentEventObserver.java

```diff
@@ -0,0 +1,116 @@
+package org.sakaiproject.assignment.impl;
+
+import java.util.Observable;
+import java.util.Observer;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.IntStream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.sakaiproject.assignment.api.AssignmentService;
+import org.sakaiproject.assignment.api.model.Assignment;
+import org.sakaiproject.assignment.api.model.AssignmentSubmission;
+import org.sakaiproject.assignment.api.model.AssignmentSubmissionSubmitter;
+import org.sakaiproject.event.api.Event;
+import org.sakaiproject.event.api.EventTrackingService;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.service.gradebook.shared.AssessmentNotFoundException;
+import org.sakaiproject.service.gradebook.shared.GradebookService;
+
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AssignmentEventObserver implements Observer {
+
+    @Setter private AssignmentService assignmentService;
+    @Setter private EventTrackingService eventTrackingService;
+    @Setter private GradebookService gradebookService;
+
+    public void init() {
+        eventTrackingService.addLocalObserver(this);
+    }
+
+    public void destroy() {
+        eventTrackingService.deleteObserver(this);
+    }
+
+    @Override
+    public void update(Observable o, Object arg) {
+        if (arg instanceof Event) {
+            Event event = (Event) arg;
+            if (event.getModify() && StringUtils.isNoneBlank(event.getEvent(), event.getContext())) {
+                switch (event.getEvent()) {
```

Comment:
If you've only got one case, why not just exit early if the event is not updateItemScore? Or just have an if statement for that one event. switch seems a bit op.

Reply:
You know people will ask for it though. You can grade a student without a submission in Assignments, and in Gradebook.

Reply:
Because this observer could be used for other events its a generic observer, and it preps it to be added on to!